impl/src/main/java/org/ehcache/UserManagedCacheBuilder.java

```diff
@@ -315,9 +376,31 @@ public T build() {
     return otherBuilder;
   }
 
-  public final UserManagedCacheBuilder<K, V, T> withCacheEvents(CacheEventDispatcher<K, V> cacheEventNotificationService) {
+  public final UserManagedCacheBuilder<K, V, T> withEventDispatcher(CacheEventDispatcher<K, V> eventNotifier) {
+    UserManagedCacheBuilder<K, V, T> otherBuilder = new UserManagedCacheBuilder<K, V, T>(this);
+    otherBuilder.eventNotifier = eventNotifier;
+    return otherBuilder;
+  }
+
+  public final UserManagedCacheBuilder<K, V, T> withOrderedEventExecutor(ExecutorService orderedExecutor) {
+    UserManagedCacheBuilder<K, V, T> otherBuilder = new UserManagedCacheBuilder<K, V, T>(this);
+    otherBuilder.orderedExecutor = orderedExecutor;
```

Comment:
By overwrite you mean that is user configured the `eventNotifier` then it is given preference over the configured executors, right ? or do we want to throw when the user configures both `eventNotifier` and executors ? 

As of now if both `eventNotifier` & executors are configured the preference is given to the provided `eventNotifier`.